src/MultiTypeRegEntry.chpl

```diff
@@ -0,0 +1,323 @@
+            var comps: map(string, string);
+            comps.add("values", "created " + st.attrib(this.array));
+            comps.add("width", this.width: string);
+            comps.add("reverse", this.reverse: string);
+            rtnMap.add("create", formatJson(comps));
+            return rtnMap;
+        }
+    }
+
+    class SegArrayRegEntry: GenRegEntry {
+        var segments: string;
+        var values: shared ArrayRegEntry;
+        var lengths: string;
+
+        proc init(segments: string, values: shared ArrayRegEntry, lengths: string) {
+            super.init(ObjType.SEGARRAY);
+            this.segments = segments;
+            this.values = values;
+            this.lengths = lengths;
+        }
+
+        proc asMap(st: borrowed SymTab): map(string, string) throws {
+            var rtnMap: map(string, string);
+            rtnMap.add("objType", this.objType: string);
+
+            var comp_create: map(string, string);
+            comp_create.add("segments", "created " + st.attrib(this.segments));
+            var val_map = this.values.asMap(st);
+            comp_create.add("values", val_map["create"]);
+            if this.lengths != "" {
+                comp_create.add("lengths", "created " + st.attrib(this.lengths));
+            }
+            rtnMap.add("create", formatJson(comp_create));
+            return rtnMap;
+        } 
+    }
+
+    class DataFrameRegEntry: GenRegEntry {
+        var idx: string; // sym_tab name of the dataframe index
+        var column_names: list(string); // list of column names
+        var columns: list(shared AbstractRegEntry); // list of sym tab names for each column+components of the column
+        proc init(idx: string, column_names: list(string), columns: list(shared AbstractRegEntry)) {
+            super.init(ObjType.DATAFRAME);
+            this.idx = idx;
+            this.column_names = column_names;
+            this.columns = columns;
+        }
+
+        proc asMap(st: borrowed SymTab): map(string, string) throws {
+            var rtnMap: map(string, string);
+            rtnMap.add("objType", this.objType: string);
+            var col_creates: map(string, string); // map column name to create statement
+            for (cname, c) in zip(this.column_names, this.columns) {
+                var gre = c: borrowed GenRegEntry;
+                var col_map: map(string, string);
+                if gre.objType == ObjType.PDARRAY || gre.objType == ObjType.STRINGS || gre.objType == ObjType.DATETIME ||
+                    gre.objType == ObjType.TIMEDELTA || gre.objType == ObjType.IPV4 {
+                    var are = gre: borrowed ArrayRegEntry;
+                    col_map = are.asMap(st);
+                }
+                else if gre.objType == ObjType.CATEGORICAL {
+                    var cre = gre: borrowed CategoricalRegEntry;
+                    col_map = cre.asMap(st);
+                }
+                else if gre.objType == ObjType.SEGARRAY {
```

Comment:
It does. However, the previous implementation of attach did not seem to work right with BitVector. I was thinking this would be added later. 

And I agree. I always forget that as an option for selects. I can go ahead and update the if to a select

Reply:
Updated